### PR TITLE
fix(activity): reduce time and memory allocated for cleaning the table

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

The query to clean up the table has suddenly been failing for many days now because of timeout. Usually that's a sign the table is too big and the row deleted do not fit in memory anymore, so PG needs to write to disk before deletion making the query much much slower.
Postgres does not allow `DELETE LIMIT` so the trick is to batch smaller and more often.  
`LIMIT 5000` can be tweaked, but usually 10K is the advisable maximum.

### CODE REVIEW

**Review by TBonnin — APPROVED** [review]
good catch. Can we make this query to fail very loudly when it does fail so we know about it?

**Comment by bodinsamuel** [comment]
Ah yes, good thinking. Adding that 👍🏻

**Comment by bodinsamuel** [comment]
Added a dedicated span (`cron.activityLogs.clean`) and a dedicated error code so we can maybe add a stable Datadog Monitor on this error. Let me know 

Screenshot 2024-02-05 at 17 25 40

File: packages/server/lib/jobs/index.ts [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
      * Delete all activity logs older than 15 days
      */
-    cron.schedule('0 * * * *', async () => {
+    cron.schedule('*/1 * * * *', async () => {
         const activityLogTableName = '_nango_activity_logs';
-        await db.knex.withSchema(db.schema()).from<ActivityLog>(activityLogTableName).where('created_at', '<', db.knex.raw("now() - interval '15 days'")).del();
+        const span = tracer.startSpan('cron.activityLogs.clean');
+        tracer.scope().activate(span, async () => {
+            try {
+                // Postgres do not allow DELETE LIMIT so we batch ourself to limit the memory footprint of this query.
```
Comment (khaliqgant) [inline packages/server/lib/jobs/index.ts]: Postgres does not -- instead of "do not"

**Review by khaliqgant — APPROVED** [review]
Nice find!

Small comment grammar change.

**Review by TBonnin — APPROVED** [review]
Thank you @bodinsamuel

**Comment by bodinsamuel** [comment]
+4M definitely not very big but that means 200K rows per day, too much to delete in one batch.

**Pull request merged**